core/src/main/java/io/substrait/relation/ExprCopyOnWriteVisitor.java

```diff
@@ -0,0 +1,366 @@
+package io.substrait.relation;
+
+import io.substrait.expression.Expression;
+import io.substrait.expression.ExpressionVisitor;
+import io.substrait.expression.FieldReference;
+import io.substrait.expression.FunctionArg;
+import java.util.List;
+import java.util.Optional;
+
+public class ExprCopyOnWriteVisitor<EXCEPTION extends Exception>
+    extends CopyOnWriteVisitor<Expression, EXCEPTION>
+    implements ExpressionVisitor<Optional<Expression>, EXCEPTION> {
+
+  private final RelCopyOnWriteVisitor<EXCEPTION> relCopyOnWriteVisitor;
+
+  public ExprCopyOnWriteVisitor(RelCopyOnWriteVisitor<EXCEPTION> relCopyOnWriteVisitor) {
+    this.relCopyOnWriteVisitor = relCopyOnWriteVisitor;
+  }
+
+  protected final RelCopyOnWriteVisitor<EXCEPTION> getRelVisitor() {
```

Comment:
Is it worth abbreviating relation here (we do elsewhere so it seems justified).  How about Expression in the name of this class?  That seems less justified.

Reply:
We use the `Rel` prefix for a number of classes already so I'm inclined to keep `RelCopyOnWriteVisitor`. We use the full `Expression` prefix elsewhere though, so I'll switch this class.